pkg/actuators/machine/machine.go

```diff
@@ -251,12 +424,34 @@ func (ms *machineScope) getClusterAddress(ctx context.Context) (map[string]int,
 	return clusterAddr, nil
 }
 
-func (ms *machineScope) reconcileMachineProviderStatus(status *string, id *string) error {
+func (ms *machineScope) findUsableInternalAddress(ctx context.Context, vmID string, excludeAddr map[string]int) (string, error) {
+
+	nicRegex := regexp.MustCompile(`^(eth|en|br\-ex).*`)
+	IPParams := ovirtC.NewVMIPSearchParams().WithIncludedInterfacePattern(nicRegex)
+	nics, err := ms.ovirtClient.GetVMIPAddresses(vmID, IPParams)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get reported devices list, reason: %w", err)
+	}
+
+	for _, ipAddressSlice := range nics {
+		for _, ipAddress := range ipAddressSlice {
+			if _, ok := excludeAddr[ipAddress.String()]; ok {
```

Comment:
This check is not necessary, GetVMIPAddresses already accepts an IP address exclusion list.

Reply:
to  exclude single address CIDR from the range, i can do something like:
`	nicRegex := regexp.MustCompile(`^(eth|en|br\-ex).*`)
	IPParams := ovirtC.NewVMIPSearchParams().WithIncludedInterfacePattern(nicRegex)
	for excludedAdrr, _ := range excludedAdrr {
		_, ipnet, err := net.ParseCIDR(fmt.Sprintf("%s/32", excludedAdrr))
		if err != nil {
			return "", errors.Wrapf(err, "failed to convert IP Address %s to CIDR", excludedAdrr)
		}
		IPParams.WithExcludedRange(*ipnet)
	}

	nics, err := ms.ovirtClient.GetVMIPAddresses(vmID, IPParams, ovirtC.ContextStrategy(ms.Context))

`